### ISSUE TITLE
Bulk density calculation in form factor script is erroneous

### ISSUE THREAD

**Issue by hsantila** [NMRLipids/MATCH#56]
The bulk density in the form factor script needs to be calculated over the entire electron profile, otherwise one may end up with a net charge in the system that manifests in the form factor as deviation at small q. The script has now been fixed, see https://github.com/NMRLipids/MATCH/pull/55/commits/42c6d170c4979d12f1d74814ab798b1b1e17185d

Every form factor curve preceeding this update and calculated with the old script should be re-calculated

**Comment by ohsOllila** [comment]
I do not fully understand this.

In my understanding, the bulk density in the form factor equation is the scattering length density in bulk water, not the average density of water, see e.g. https://doi.org/10.1007/s00232-010-9254-5. X-ray experiment is measuring the contrast of the electron density with respect to the electron density in bulk water. The total electron density in the lipid region does not have to be the same as in bulk water, therefore the form factor does not have to be zero at zero q.

In conclusion, I think that the bulk density should be the density in bulk water, which was taken as the last point in the density profile in the code as bulkDENS=$(tail -n 1 electronDENSITY.xvg | awk '{print $2}').

**Comment by ohsOllila** [comment]
I have now copied calc_FORM_FACTOR.sh script to the script folder and use this path in scripts from now on. The version in the script folder uses the bulk water density, not average.

**Comment by hsantila** [comment]
You are right. We confused with electron density with charge density and my statement above does not make any sense.

However, using one point for the density estimate makes the form factor calculation very prone to noise in that point. I have now implemented a version where the density is averaged from the 2 water layers at the box edges and made a pull request.

**Comment by markussmiettinen** [comment]
There seems to be three versions of code:

1. the old one at /scripts
2. the new one by @hsantila in the /scratch/report
3. another new one (called center_n_calcFF.sh) that allows multiple molecule types at /scratch/scriptsbyhsantila

I guess we should replace 1. and 2. by 3. and remove the others, @ohsOllila ?

**Comment by ohsOllila** [comment]
There is a new form factor code in the databank: https://github.com/NMRLipids/Databank/blob/main/Scripts/AnalyzeDatabank/calc_FormFactors.py. Therefore this issue is outdated and I will close it.
